modules/hats_in_a_book/hats_in_a_book.module

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * @file
+ * Places page objects in a single book.
+ */
+
+/**
+ * Implements hook_ten_million_with_a_hat_also_do_these_things().
+ */
+function hats_in_a_book_ten_million_with_a_hat_also_do_these_things() {
+  global $user;
+  cache_clear_all('hats_in_a_book_current_page', 'cache');
+  $tuque = islandora_get_tuque_connection();
+  $book = $tuque->repository->constructObject('book');
+  $book->label = "Book hat";
+  $book->owner = $user->name;
+  $book->models = (array) 'islandora:bookCModel';
+  $book->relationships->add(FEDORA_RELS_EXT_URI, 'isMemberOfCollection', 'islandora:bookCollection');
+  $tuque->repository->ingestObject($book);
+  cache_set('hats_in_a_book_current_page', 1);
+  return array(
+    array(
+      'callback' => 'hats_in_a_book_add_to_book',
+      'args' => array($book->id),
+      'when' => 'between_ingests',
+    ),
+    array(
+      'when' => 'after_batch',
+      'callback' => 'hats_in_a_book_after_batch_callback',
+      'args' => array($book->id),
+    ),
+  );
+}
+
+/**
+ * Adds the page to a book, if necessary.
+ *
+ * @param AbstractObject $object
+ *   The object to add to the book if it's a page.
+ * @param string $book
+ *   The PID of the book to add the pages to.
+ */
+function hats_in_a_book_add_to_book(AbstractObject $object, $book) {
+  if (in_array('islandora:pageCModel', $object->models)) {
+    // Add the page to the book.
+    module_load_include('inc', 'islandora_paged_content', 'includes/utilities');
+    $current_page = cache_get('hats_in_a_book_current_page');
+    $current_page = $current_page->data;
+    islandora_paged_content_set_relationship($object->relationships, ISLANDORA_RELS_EXT_URI, 'isPageOf', $book);
```

Comment:
Should call https://github.com/Islandora/islandora_paged_content/blob/7.x/includes/utilities.inc#L1127